husc/pathutils.py

```diff
@@ -0,0 +1,34 @@
+import os
+from fnmatch import fnmatch
+
+def all_matching_files(path, glob='*.TIF', full=True, sort=True):
+    """Return list of files in directory with specified extension.
+
+    Note: This function recurses all subdirectories.
+
+    Parameters
+    ----------
+    path : string
+        A directory containing files.
+    glob : string
+        Pattern used to match files. eg. '*.TIF' will return all
+        TIF files.
+    full : bool, optional
+        Whether or not to return files with the path included.
+    sort : bool, optional
+        Whether or not to sort the list of files before returning them.
+    """
+    fns = []
+    if full is True:
```

Comment:
Move the if to the inside loop to avoid repetition. I know it's slower but it's marginal and will massively aid readability and maintainability